### PR TITLE
Zero precision in timestamp/datetime #4784

### PR DESCRIPTION
**Pull request by OlivierCavadenti**
- Fixes https://github.com/knex/knex/issues/4784
- Remove duplicate (actually timestamp == datetime implementation in PG) and so correct datetime precision = 0 too.

### CODE REVIEW

File: lib/dialects/postgres/schema/pg-columncompiler.js [hunk 1]
```diff
@@ -72,22 +72,13 @@ class ColumnCompiler_PG extends ColumnCompiler {
       useTz = !withoutTz;
     }
     useTz = typeof useTz === 'boolean' ? useTz : true;
-    precision = precision ? '(' + precision + ')' : '';
+    precision = precision !== undefined ? '(' + precision + ')' : '';
```
Comment (kibertoad) [inline lib/dialects/postgres/schema/pg-columncompiler.js:75]: can we also check for null here?

File: lib/dialects/postgres/schema/pg-columncompiler.js [hunk 2]
```diff
@@ -72,22 +72,13 @@ class ColumnCompiler_PG extends ColumnCompiler {
...
-    if (isObject(withoutTz)) {
-      ({ useTz, precision } = withoutTz);
-    } else {
-      useTz = !withoutTz;
-    }
-    useTz = typeof useTz === 'boolean' ? useTz : true;
-    precision = precision ? '(' + precision + ')' : '';
-
-    return `${useTz ? 'timestamptz' : 'timestamp'}${precision}`;
+    return this.datetime(withoutTz, precision);
```
Comment (kibertoad) [inline lib/dialects/postgres/schema/pg-columncompiler.js:81]: great catch!

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
@OlivierCavadenti If you think we need to release it as a hotfix, we should also backport it to 0.95 branch.

**Comment by OlivierCavadenti** [comment]
> @OlivierCavadenti If you think we need to release it as a hotfix, we should also backport it to 0.95 branch.

yes we can. do you have any conventation (naming, ...) to make the backport ?

**Comment by kibertoad** [comment]
@OlivierCavadenti I don't think we have one. Also feel free to merge approved PRs at your convenience!

**Pull request merged**